Remove unchecked feature

vbkaisetsu opened the pull request:
This PR removes `unchecked` feature from this crate and adds some `unsafe` functions.

The current `unchecked` feature disables `Dictionary::read()`. Features may add functions, but should not remove. Also, `Dictionary::read()` will not be shown in docs.rs because the document will be built by the following command:
```
RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --all-features
```

This branch adds the `need_check` flag to `Dictionary` and switches functions according to that flag.

kampersanda commented on benchmark/Cargo.toml:
```diff
@@ -7,7 +7,6 @@ edition = "2021"
 [features]
 default = []
-unchecked = ["vibrato/unchecked"]
 
```
```suggestion
unchecked = []
```

Could you keep the `unchecked` feature in the benchmark tool since it is used to specify `Dictionary::read_unchecked()`?

kampersanda commented on prepare/Cargo.toml:
```diff
@@ -7,7 +7,6 @@ edition = "2021"
 [features]
 default = []
-unchecked = ["vibrato/unchecked"]
 
```
```suggestion
unchecked = []
```
ditto

kampersanda commented on tokenize/Cargo.toml:
```diff
@@ -7,7 +7,6 @@ edition = "2021"
 [features]
 default = []
-unchecked = ["vibrato/unchecked"]
 
```
```suggestion
unchecked = []
```
ditto

vbkaisetsu commented:
I added `unchecked` feature of the benchmark, but I removed calling unchecked functions from `prepare` and `tokenize` commands.

The `unchecked` feature is very dangerous.

* These commands read untrusted files from the outside, and not checking files is not desirable from a security standpoint.
* I do not think that the processing time of the prepare command is a major issue.
* In the tokenize command, standard input/output is expected to occupy most of the processing time, so I think that the speed improvement is limited.

kampersanda reviewed (commented):
Okay, I agree.

Lastly, could you remove the section [Faster tokenization](https://github.com/daac-tools/vibrato/tree/remove-unchecked-feature#faster-tokenization) and move the description into the section [Benchmark](https://github.com/daac-tools/vibrato/tree/remove-unchecked-feature#benchmark)?

kampersanda reviewed (approved):
Thank you very much!
